simulator/reset/reset.go

```diff
@@ -39,17 +44,33 @@ func NewResetService(
 	deleteServices map[string]DeleteService,
 	deleteServicesForNamespacedResources map[string]DeleteServicesForNamespacedResources,
 	schedService SchedulerService,
+	ctrlerManager CtrlerManager,
 ) *Service {
 	return &Service{
 		client:                               client,
 		deleteServices:                       deleteServices,
 		deleteServicesForNamespacedResources: deleteServicesForNamespacedResources,
 		schedService:                         schedService,
+		ctrlerManager:                        ctrlerManager,
 	}
 }
 
 // Reset cleans up all resources and scheduler configuration.
 func (s *Service) Reset(ctx context.Context) error {
+	if err := s.deletek8sResources(ctx); err != nil {
+		return xerrors.Errorf("delete k8s Resources: %w", err)
+	}
+
+	// If the ResetController fail, klog.Fatalf be called internally.
+	s.ctrlerManager.ResetController()
```

Comment:
I don't think we need to restart controllers and scheduler when `Reset` is called.

Reply:
I think what @jonyhy96 mean is that we need to remove deployment and replicaset when `Reset` is called.

But, I'll refactor reset service either way, so, you can just forget the reset service for now.
https://github.com/kubernetes-sigs/kube-scheduler-simulator/issues/218